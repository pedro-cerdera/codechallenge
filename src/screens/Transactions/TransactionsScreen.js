import React from 'react';
import {Text, View, StyleSheet, FlatList} from 'react-native';
import {Icon, TransactionList} from '../../components';
import styleguide from '../../styleguide';
import {useSelector} from 'react-redux';
import {maskMoney} from '../../helpers';

const TransactionsScreen = () => {
  const {transactions} = useSelector((store) => ({
    transactions: store.TransactionReducer.transactions,
  }));

  return (
    <View
      testID={'transactions-screen'}
      style={{flex: 1, backgroundColor: 'white'}}>
      <TransactionList transactions={transactions} />
    </View>
  );
};

export default TransactionsScreen;
