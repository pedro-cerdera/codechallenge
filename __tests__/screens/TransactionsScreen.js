import React from 'react';
import {render} from 'react-native-testing-library';
import {useSelector} from 'react-redux';
import TransactionsScreen from '../../src/screens/Transactions/TransactionsScreen';

jest.mock(
  '../../src/components/TransactionList/TransactionList',
  () => 'TransactionList',
);
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

const mockAppState = {
  TransactionReducer: {
    ammount: 2000,
    transactions: [
      {
        value: -100,
        description: 'Transação 1',
      },
      {
        value: 2100,
        description: 'Transação 2',
      },
    ],
    isLoading: false,
    error: null,
  },
};

describe('Testing TransactionsScreen', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => {
      return callback(mockAppState);
    });
  });
  afterEach(() => {
    useSelector.mockClear();
  });

  test('Render all home screen components correctly', async () => {
    const {getByTestId} = render(<TransactionsScreen />);
    expect(getByTestId('transaction-list').props.transactions).toBe(
      mockAppState.TransactionReducer.transactions,
    );
  });

  test('renders correctly', () => {
    const tree = render(<TransactionsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
