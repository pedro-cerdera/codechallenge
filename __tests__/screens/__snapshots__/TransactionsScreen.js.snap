// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TransactionsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  testID="transactions-screen"
>
  <TransactionList
    testID="transaction-list"
    transactions={
      Array [
        Object {
          "description": "Transação 1",
          "value": -100,
        },
        Object {
          "description": "Transação 2",
          "value": 2100,
        },
      ]
    }
  />
</View>
`;
